src/utils/check.ts

```diff
@@ -1,3 +1,169 @@
+import { AxePuppeteer } from '@axe-core/puppeteer';
+import puppeteer from 'puppeteer';
+import * as fs from 'fs';
+import { exec } from 'child_process';
+import { getConfig } from '@/helpers/config';
+
+const retry = async (
+  url: string,
+  resolve: any,
+  retries: number = 5,
+  delay: number = 5000,
+) => {
+  for (let attempt = 0; attempt < retries; attempt++) {
+    try {
+      await new Promise((res) => setTimeout(res, delay));
+
+      const response = await fetch(url);
+
+      if (response.ok) {
+        console.log('Build is up and running.');
+        resolve();
+        return;
+      } else {
+        console.log(
+          `Serving attempt ${attempt + 1} failed: Server is not up yet. Retrying...`,
+        );
+      }
+    } catch (error) {
+      console.log(
+        `Serving attempt ${attempt + 1} failed: ${error}. Retrying...`,
+      );
+    }
+  }
+
+  throw `Server failed to start after ${retries} attempts.`;
+};
+
+const build = async (cmd: string) =>
+  new Promise<void>((resolve, reject) => {
+    exec(cmd, (error, stdout, stderr) => {
+      if (error) {
+        return reject(`Build error: ${error}`);
+      }
+      if (stderr) {
+        return reject(`Build stderr: ${stderr}`);
+      }
+      console.log(`Build stdout: ${stdout}`);
+      resolve();
+    });
+  });
+
+const serveBuild = async (cmd: string) =>
+  new Promise<{ serverProcess: any }>((resolve, reject) => {
+    const serverProcess = exec(cmd, (error, stdout, stderr) => {
+      if (error) {
+        return reject(`Serve error: ${error}`);
+      }
+      if (stderr) {
+        return reject(`Serve stderr: ${stderr}`);
+      }
+      console.log(`Serve stdout: ${stdout}`);
+    });
+
+    const url = 'http://localhost:3000/';
+    retry(url, () => resolve({ serverProcess })).catch(reject);
+  });
+
+const stopServer = (serverProcess: any) => {
+  try {
+    process.kill(serverProcess.pid + 1, 'SIGTERM');
+    process.exit(0);
+  } catch (error) {
+    throw `Error stopping the server: ${error}`;
+  }
+};
+
 export async function check(flag: boolean) {
-  console.log(`check${flag ? ' --report' : ''}`);
+  let serverProcess;
+
+  try {
+    const _config = await getConfig();
+    switch (_config?.framework) {
+      case 'React':
+        await build('npm run build');
+        ({ serverProcess } = await serveBuild('npx serve -s build -p 3000'));
+        break;
+      case 'Vue':
+        await build('npx vite build');
+        ({ serverProcess } = await serveBuild('npx serve -s dist -p 3000'));
+        break;
+      case 'Angular':
+        // await build('npx ng build');
```

Comment:
is this meant to be commented out

Reply:
Yup, angular handle the build for us with serve cmd